### ISSUE TITLE
Unclear error message when OSX installation fails

### ISSUE THREAD

**Issue by maurelian** [iamdefinitelyahuman/py-solc-x#26]
This is an issue from an upstream tool: https://github.com/dmuhs/mythx-cli/issues/34, which I've been asked to report here:

* MythX CLI version: 0.2.1
* Python version: 3.7.5
* Operating System: 10.15.1 Catalina 

### Description
```
mythx analyze --mode quick contracts/**/*.sol
Unsupported macOS version.
We only support Mavericks, Yosemite, El Capitan, Sierra, High Sierra and Mojave.
Usage: mythx analyze [OPTIONS] [TARGET]...

Error: Error installing solc version v0.5.10: Command '['sh', '/var/folders/vk/01zl87497jx6bq66fhhhd9zm0000gn/T/py-solc-x-tmp/solidity_0.5.10/scripts/install_deps.sh']' returned non-zero exit status 1.
```

**Comment by iamdefinitelyahuman** [comment]
This looks like a compatibility issue between solc and OSX 10.15.  Seems it was fixed in https://github.com/ethereum/solidity/pull/7367 and first included in solc v0.5.12.

If you can install solc v0.5.10 manually via brew, py-solc-x will use that version rather than attempting to install it's own copy, and you can avoid this issue. Unfortunately I don't have a copy of OSX 10.15 available to experiment with, so not certain if this is actually an option.

**Comment by maurelian** [comment]
Yeah, fortunately Solidity does support previous version using brew. 

To leave a breadcrumb, instructions to do so are towards the end of this section: https://solidity.readthedocs.io/en/v0.6.0/installing-solidity.html#binary-packages

**Comment by iamdefinitelyahuman** [comment]
Good idea - I'll improve the verbosity when an OSX install fails, to include mention of installing via brew.  Thanks!

**Comment by iamdefinitelyahuman** [comment]
I took a closer look and realized I can just log a warning when the dependency script fails, there's no need to raise an exception. And I've improved the verbosity so users aren't left in the dark when it does fail.

Fix is on the [v0.6.1](https://github.com/iamdefinitelyahuman/py-solc-x/compare/v0.6.1?expand=1) branch. I'm gonna deal with issues re: solc 0.6.0 and then will push a new version to pypi.

Thanks again for pointing this out!
